### ISSUE TITLE
Can not run Configured task

### ISSUE THREAD

**Issue by RomanNikitenko** [theia-ide/theia#5064]
### Description
Task configuration can not be found for running after configuring 'detected' task.

### Reproduction Steps
1. Configure a 'detected' task
2. Try to run the 'configured' task

You can see error in the browser console: "task ERROR Can't get task launch configuration for label: build"

Please see https://youtu.be/8TNbMWVv2y8

**Diagnostics:**

As far as I can see [the cause of the problem](https://github.com/theia-ide/theia/commit/0383b1b13b156cde4b5aea2c184b815455733c36#diff-65aa48cf1be5fc5d2c202882762b24c4R237):
- source folder is used as key for [saving configurations of tasks to map](https://github.com/theia-ide/theia/blob/c2e23083b905130276238477646b72d298d47d0d/packages/task/src/browser/task-configurations.ts#L214-L215) from a config file
- `_source` is used to [get task](https://github.com/theia-ide/theia/blob/c2e23083b905130276238477646b72d298d47d0d/packages/task/src/browser/task-service.ts#L215) configuration [at running](https://github.com/theia-ide/theia/blob/c2e23083b905130276238477646b72d298d47d0d/packages/task/src/browser/quick-open-task.ts#L271)
- but [source of a task](https://github.com/theia-ide/theia/blob/c2e23083b905130276238477646b72d298d47d0d/packages/task/src/browser/task-configurations.ts#L237) (not source folder) is set as `_source` when configuration is read from a config file

In other words path to the source folder is used to save configuration, but type of a provider for 'detected' task is used to find this configuration for running.

**Comment by akosyakov** [comment]
cc @elaihau

**Comment by RomanNikitenko** [comment]
The behavior was changed: after configuring of `detected` task, this one is not present in `configured tasks` section, so I can not reproduce the issue.
The issue can be closed, if the new behavior is correct.

I created the issue https://github.com/theia-ide/theia/issues/5747 to clarify and discuss the new behavior. 
@elaihau
